agent/config/builder.go

```diff
@@ -1004,64 +1004,65 @@ func (b *builder) build() (rt RuntimeConfig, err error) {
+		RPCConfig:                        consul.RPCConfig{EnableStreaming: boolValWithDefault(c.RPC.EnableStreaming, serverMode)},
+		RaftProtocol:                     intVal(c.RaftProtocol),
+		RaftSnapshotThreshold:            intVal(c.RaftSnapshotThreshold),
+		RaftSnapshotInterval:             b.durationVal("raft_snapshot_interval", c.RaftSnapshotInterval),
+		RaftTrailingLogs:                 intVal(c.RaftTrailingLogs),
+		ReconnectTimeoutLAN:              b.durationVal("reconnect_timeout", c.ReconnectTimeoutLAN),
+		ReconnectTimeoutWAN:              b.durationVal("reconnect_timeout_wan", c.ReconnectTimeoutWAN),
+		RejoinAfterLeave:                 boolVal(c.RejoinAfterLeave),
+		RetryJoinIntervalLAN:             b.durationVal("retry_interval", c.RetryJoinIntervalLAN),
+		RetryJoinIntervalWAN:             b.durationVal("retry_interval_wan", c.RetryJoinIntervalWAN),
+		RetryJoinLAN:                     b.expandAllOptionalAddrs("retry_join", c.RetryJoinLAN),
+		RetryJoinMaxAttemptsLAN:          intVal(c.RetryJoinMaxAttemptsLAN),
+		RetryJoinMaxAttemptsWAN:          intVal(c.RetryJoinMaxAttemptsWAN),
+		RetryJoinWAN:                     b.expandAllOptionalAddrs("retry_join_wan", c.RetryJoinWAN),
+		SegmentName:                      stringVal(c.SegmentName),
+		Segments:                         segments,
+		SegmentLimit:                     intVal(c.SegmentLimit),
+		SerfAdvertiseAddrLAN:             serfAdvertiseAddrLAN,
+		SerfAdvertiseAddrWAN:             serfAdvertiseAddrWAN,
+		SerfAllowedCIDRsLAN:              serfAllowedCIDRSLAN,
+		SerfAllowedCIDRsWAN:              serfAllowedCIDRSWAN,
+		SerfBindAddrLAN:                  serfBindAddrLAN,
+		SerfBindAddrWAN:                  serfBindAddrWAN,
+		SerfPortLAN:                      serfPortLAN,
+		SerfPortWAN:                      serfPortWAN,
+		ServerMode:                       serverMode,
+		ServerName:                       stringVal(c.ServerName),
+		ServerPort:                       serverPort,
+		Services:                         services,
+		SessionTTLMin:                    b.durationVal("session_ttl_min", c.SessionTTLMin),
+		SkipLeaveOnInt:                   skipLeaveOnInt,
+		StartJoinAddrsLAN:                b.expandAllOptionalAddrs("start_join", c.StartJoinAddrsLAN),
+		StartJoinAddrsWAN:                b.expandAllOptionalAddrs("start_join_wan", c.StartJoinAddrsWAN),
+		TaggedAddresses:                  c.TaggedAddresses,
+		TranslateWANAddrs:                boolVal(c.TranslateWANAddrs),
+		TxnMaxReqLen:                     uint64Val(c.Limits.TxnMaxReqLen),
+		UIConfig:                         b.uiConfigVal(c.UIConfig),
+		UnixSocketGroup:                  stringVal(c.UnixSocket.Group),
+		UnixSocketMode:                   stringVal(c.UnixSocket.Mode),
+		UnixSocketUser:                   stringVal(c.UnixSocket.User),
+		Watches:                          c.Watches,
+		AutoReloadConfigCoalesceInterval: 1 * time.Second,
```

Comment:
This is not an exposed config yet but I thought we can expose it. in the future if we have requests for it.